runners/direct-java/src/main/java/org/apache/beam/runners/direct/DirectExecutionContext.java

```diff
@@ -48,22 +48,37 @@ public DirectExecutionContext(
     this.watermarks = watermarks;
   }
 
-  @Override
-  protected DirectStepContext createStepContext(String stepName, String transformName) {
-    return new DirectStepContext(this, stepName, transformName);
+  private DirectStepContext createStepContext(String stepName, String transformName) {
+    return new DirectStepContext(stepName, transformName);
+  }
+
+  /**
+   * Returns the {@link StepContext} associated with the given step.
+   */
+  public DirectStepContext getStepContext(String stepName, String transformName) {
+    final String finalStepName = stepName;
```

Comment:
How do the `stepName` and `transformName` relate? Does the direct runner need both?